src/entry.c

```diff
@@ -976,6 +976,52 @@ vstumpless_set_entry_message( struct stumpless_entry *entry,
   return entry;
 }
 
+const char *
+stumpless_get_entry_procid( const struct stumpless_entry *entry ) {
+  char *procid;
+  
+  VALIDATE_ARG_NOT_NULL( entry );
+
+  procid = alloc_mem( sizeof(char) * 129 );
+
+  if( procid == NULL ) {
+    raise_memory_allocation_failure();
+    return NULL;
+  }
+
+  lock_entry( entry );
+  
+  if( entry->procid_override == false ) {
+    sprintf( procid, "%d" , config_getpid() );
```

Comment:
Avoid `sprintf` unless the formatting is non-trivial. In this case, you can use the code from `strbuilder_append_positive_int` or write a new helper in `include/private/inthelper.h`.